### ISSUE TITLE
support remote icalendar url loads

### ISSUE THREAD

**Issue by tacman** [fullcalendar/fullcalendar#7476]
### Checklist

- [X] I've already searched through [existing tickets](https://github.com/fullcalendar/fullcalendar/issues)
- [X] Other people will find this feature useful

### Is this feature for a specific connector (React/Angular/etc)?

No connector in particular

### Feature Description

https://fullcalendar.io/docs/icalendar

Can you change the example to show a remote url?  I keep getting a CORS error, even with public urls

```js
import dayGridPlugin from '@fullcalendar/daygrid'
import iCalendarPlugin from '@fullcalendar/icalendar'
var url = 'https://www.co.midland.tx.us/common/modules/iCalendar/iCalendar.aspx?catID=22&feed=calendar ';
var calendar = new Calendar(calendarEl, {
  plugins: [dayGridPlugin, iCalendarPlugin],
  events: {
    url: url,
    format: 'ics'
  }
})

calendar.render()
```

```
micro-calendar.wip/:1 Access to fetch at 'https://www.co.midland.tx.us/common/modules/iCalendar/iCalendar.aspx?catID=22&feed=calendar' from origin 'https://micro-calendar.wip' has been blocked by CORS policy: No 'Access-Control-Allow-
Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

(index):182     GET https://www.co.midland.tx.us/common/modules/iCalendar/iCalendar.aspx?catID=22&feed=calendar net::ERR_FAILED 200 (OK)
```

Can the underlying fetch call be made with the appropriate headers?  That is, to set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

**Comment by acerix** [comment]
I don't believe there's a way to bypass CORS from the client side, it would need to be allowed by the server.  

I think it would be useful to have a demo using a public URL in the docs though.  I found a feed that allows it and made a demo here:

https://stackblitz.com/edit/github-t6aomg?file=src%2FDemoApp.jsx

**Comment by tacman** [comment]
The demo at https://stackblitz.com/edit/github-t6aomg?file=src%2FDemoApp.jsx no longer works (CORS error).

**Comment by haegar33** [comment]
Would be great to see a full example (html + js files )?? I cannot get this working. Always complaining that imports only work on "top level" module..?

**Comment by lbell** [comment]
> As a workaround, you can create a backend endpoint from your own domain to serve the ics data from another domain.

@justin-hackin Can you expand on this?

**Comment by lbell** [comment]
Voting for accepting an ICS formatted string as events input, rather than just a URL (which throws CORS error).

I can get the data to my server, and it would be straightforward to hand if off to FullCalendar as a text string.

**Comment by justin-hackin** [comment]
> > As a workaround, you can create a backend endpoint from your own domain to serve the ics data from another domain.
> 
> @justin-hackin Can you expand on this?



> > As a workaround, you can create a backend endpoint from your own domain to serve the ics data from another domain.
> 
> @justin-hackin Can you expand on this?

Your server can fetch the data without CORS errors, then serve it from your own domain where your control the CORS policy

**Comment by tacman** [comment]
Can the library accept a JSON _string_ (or a js array of objects)?  That way the caller can get the data and pass it, rather than passing the url to the library.

A working example would rock, CORS is a pain to deal with.

**Comment by justin-hackin** [comment]
FYI, this library https://github.com/schedule-x/schedule-x now has support for ical data as a string as per the docs here https://schedule-x.dev/docs/calendar/plugins/ical
